### PR TITLE
wgsl: Migrate AF 'fract' to use look up table

### PR DESCRIPTION
**Pull request by zoddicus**
This updates the fract tests to be compliant with a recent change to
the spec, https://github.com/gpuweb/gpuweb/pull/4541.

This also includes some refactoring of the existing generation code
for better reuse, removing some spurious zeros in the output, and
avoiding potentially ambigious fractional values, by changing them to
hex values.

Fixes https://github.com/gpuweb/cts/issues/3672

<hr>

**Requirements for PR author:**

- [ ] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [ ] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [ ] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by dneto0 — APPROVED** [review]
Thanks!

File: src/webgpu/shader/execution/expression/binary/af_data.ts [hunk 1]
```diff
@@ -42,58 +73,58 @@ export const kAdditionRawValues:  readonly RawBinaryData[] = [
   { lhs: -10.0, rhs: kValue.f64.negative.min, expected: [ kValue.f64.negative.min ] },
   { lhs: -10.0, rhs: -10.0, expected: [ -20 ] },
   { lhs: -10.0, rhs: -1.0, expected: [ -11 ] },
-  { lhs: -10.0, rhs: -0.125, expected: [ -10.125 ] },
+  { lhs: -10.0, rhs: -0.125, expected: [ reinterpretU64AsF64(0xc024400000000000n) /* ~-10.125 */ ] },
```
[the remark this answers is not in the document]
Comment (zoddicus) [reply src/webgpu/shader/execution/expression/binary/af_data.ts]: Done

**Comment by zoddicus** [comment]
@dneto0 Can I get a review on the dependeny PR, https://github.com/gpuweb/cts/pull/3667

**Comment by zoddicus** [comment]
> @dneto0 Can I get a review on the dependeny PR, #3667

Dan reviewed

**Pull request merged**